packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/TableCellSelection.ts

```diff
@@ -188,20 +188,27 @@ export default class TableCellSelection implements EditorPlugin {
                 clonedVTable.selection = this.tableRange;
                 removeCellsOutsideSelection(clonedVTable);
                 clonedVTable.writeBack();
-
                 event.range.selectNode(clonedTable);
 
                 if (event.isCut) {
-                    forEachSelectedCell(this.vTable, cell => {
-                        if (cell?.td) {
-                            deleteNodeContents(cell.td, this.editor);
-                        }
-                    });
+                    this.deleteTableColumns(selection.table, selection.coordinates);
                 }
             }
         }
     }
 
+    private deleteTableColumns(table: HTMLTableElement, selection: TableSelection) {
+        const isWholeColumnSelection = table.rows.length - 1 === selection.lastCell.y;
+
+        if (isWholeColumnSelection) {
```

Comment:
What if whole row is selected? we should handle both cases

Reply:
The word online just handles whole columns or whole table. If a whole row is or not whole columns is selected, the editor does dot remove the cells.